layers/cip.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2017 Google, Inc. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file in the root of the source
+// tree.
+
+package layers
+
+import (
+	"encoding/binary"
+	"errors"
+
+	"github.com/google/gopacket"
+)
+
+const (
+	cipBasePacketLen int = 2
+)
+
+var (
+	ErrCIPDataTooSmall = errors.New("CIP packet data truncated")
+)
+
+type CIP struct {
+	BaseLayer
+	Response         bool     // false if request, true if response
+	ServiceID        byte     // The service specified for the request
+	ClassID          uint16   // request only
+	InstanceID       uint16   // request only
+	Status           byte     // Response only
+	AdditionalStatus []uint16 // Response only
```

Comment:
I notice that Status is 1 byte, but AdditionalStatus is a set of 2-byte values... just want to make sure this was intentional :)

Reply:
Yep, page 23 of https://www.rockwellautomation.com/resources/downloads/rockwellautomation/pdf/sales-partners/technology-licensing/eipexp1_2.pdf defines "Additional status" to be a variable-length array of 16-bit words.